server/channels/app/post_metadata.go

```diff
@@ -615,7 +615,7 @@ func (a *App) getLinkMetadata(c request.CTX, requestURL string, timestamp int64,
 	if looksLikeAPermalink(requestURL, a.GetSiteURL()) && *a.Config().ServiceSettings.EnablePermalinkPreviews {
 		referencedPostID := requestURL[len(requestURL)-26:]
 
-		referencedPost, appErr := a.GetSinglePost(referencedPostID, false)
+		referencedPost, appErr := a.GetPostIfAuthorized(c, referencedPostID, c.Session(), false)
```

Comment:
Why was this changed? Doesn't this break websocket broadcasts?

Reply:
I mean getLinkMetadata is called from places like createPost and updatePost to populate the post metadata, and then that post is broadcast to the channel. The author of the post might not have permission to view a particular referencedPost, but some of the recipients of the broadcast still could.